src/main/java/Calculator.java

```diff
@@ -1,72 +1,171 @@
 import java.math.BigDecimal;
 
+/**
+ * The Calculator class provides methods to perform arithmetic operations on MoneyValue objects.
+ * It supports addition, subtraction, multiplication, and division of MoneyValue instances.
+ * The class ensures that the operations are performed in the same currency, using a ConverterClient
+ * if necessary to handle currency conversions.
+ */
 public class Calculator {
     private MoneyValueClient mv_;
     private ConverterClient cv_;
 
+    /**
+     * Constructs a Calculator with the specified MoneyValue and Converter clients.
+     * The MoneyValue client represents the current monetary value and currency,
+     * while the Converter client is used to convert between different currencies if needed.
+     *
+     * @param mv the MoneyValue client representing the current monetary value
+     * @param cv the Converter client used for currency conversion
+     */
     public Calculator(MoneyValue mv, Converter cv) {
         this.mv_ = mv;
         this.cv_ = cv;
     }
 
-    public MoneyValueClient getMoneyValueClient() { return this.mv_; }
-    public ConverterClient getConverter() { return this.cv_; }
+    /**
+     * Returns the current MoneyValue client.
+     * This client contains the monetary value and its associated currency.
+     *
+     * @return the MoneyValue client representing the current monetary value
+     */
+    public MoneyValueClient getMoneyValueClient() {
+        return this.mv_;
+    }
+
+    /**
+     * Returns the Converter client.
+     * The Converter client is responsible for converting monetary values between different currencies.
+     *
+     * @return the Converter client used for currency conversion
+     */
+    public ConverterClient getConverter() {
+        return this.cv_;
+    }
+
+    /**
```

Comment:
First sentence is unnecessary imo. Is the fact that an exception can be thrown not default seeable?

Reply:
Imo just the @throws comment should be enough.